src/main/java/com/palantir/gradle/graal/GradleGraalPlugin.java

```diff
@@ -45,60 +43,39 @@
 public class GradleGraalPlugin implements Plugin<Project> {
 
     /** Location to cache downloaded and extracted GraalVM tooling. */
-    public static final Path CACHE_DIR =
-            Paths.get(System.getProperty("user.home"), ".gradle", "caches", "com.palantir.graal");
+    static final Path CACHE_DIR = Optional.ofNullable(System.getProperty("com.palantir.graal.cache.dir"))
+            .map(Paths::get)
+            .orElse(Paths.get(System.getProperty("user.home"), ".gradle", "caches", "com.palantir.graal"));
 
-    public static final String TASK_GROUP = "Graal";
+    static final String TASK_GROUP = "Graal";
 
     @Override
     public final void apply(Project project) {
+        project.getPluginManager().apply("java");
         GraalExtension extension = project.getExtensions().create("graal", GraalExtension.class, project);
 
-        DownloadGraalTask downloadGraal = project.getTasks().create(
-                "downloadGraalTooling", DownloadGraalTask.class, task -> {
-                    task.configure(extension.getGraalVersion(), extension.getDownloadBaseUrl());
-                });
+        TaskProvider<DownloadGraalTask> downloadGraal = project.getTasks().register(
+                "downloadGraalTooling",
+                DownloadGraalTask.class,
+                extension);
 
-        ExtractGraalTask extractGraal = project.getTasks().create(
-                "extractGraalTooling", ExtractGraalTask.class, task -> {
-                    task.dependsOn(downloadGraal);
-                    task.configure(asProvider(() -> downloadGraal.getOutput().toFile()), extension.getGraalVersion());
-                });
+        TaskProvider<ExtractGraalTask> extractGraal = project.getTasks().register(
+                "extractGraalTooling",
+                ExtractGraalTask.class,
+                extension);
```

Comment:
if removing the injected param, can pass the configure action in here as the last arg